File: web/classroommanagement/EMS.Web/ClientApp/src/app/feature/class/class-search/class-search-request.ts
import { Paging } from 'src/app/shared/models/api/paging';

export class ClassSearchRequest extends Paging  {
    CourseId: number;
    Lesson: number;
    Week: number;
    Session: string;
    Card: string;
    Fictionality: string;
    Genre: string;
    Title: string;
    Classwork: string;
    Homework: string;
    LessonPlan: string;
    Phonics: string;
}

File: api/SilverLeaf.Entities/ViewModels/Requests/ClassSearchRequest.cs
using SilverLeaf.Common.Models;

namespace SilverLeaf.Entities.ViewModels.Requests
{
    public class ClassSearchRequest : PagingRequest
    {
        public ClassSearchRequest()
        {

        }

        public ClassSearchRequest(PagingRequest pagingRequest)
        {
            From = pagingRequest.From;
            Sort = pagingRequest.Sort;
            Fields = pagingRequest.Fields;
            Size = pagingRequest.Size;
        }

        public int CourseId { get; set; }

        public int Id { get; set; }

        public string Name { get; set; }

        public int Lesson { get; set; }

        public int Week { get; set; }

        public string Session { get; set; }

        public string Card { get; set; }

        public string Fictionality { get; set; }

        public string Genre { get; set; }

        public string Title { get; set; }

        public bool IsActive => true;
    }
}
